packages/node-core/src/indexer/multiChainRewind.service.test.ts

```diff
@@ -0,0 +1,127 @@
+const genBlockTimestamp = (height: number) => (1740100000 + height) * 1000;
+const genBlockDate = (height: number) => new Date(genBlockTimestamp(height));
+
+const testSchemaName = 'test_multi_chain_rewind';
+const schema = buildSchemaFromString(`
+  type Account @entity {
+    id: ID! # Account address
+    balance: Int
+  }
+`);
+
+describe('Check whether the db store and cache store are consistent.', () => {
+  let sequelize: Sequelize;
+  let storeService: StoreService;
+  let multiChainRewindService: MultiChainRewindService;
+
+  // Mock IBlockchainService
+  const mockBlockchainService = {
+    getHeaderForHeight: jest.fn((height: number) => ({
+      blockHeight: height,
+      timestamp: genBlockDate(height),
+      blockHash: `hash${height}`,
+      parentHash: height > 0 ? `hash${height - 1}` : '',
+    })),
+  };
+
+  beforeAll(async () => {
+    sequelize = new Sequelize(
+      `postgresql://${option.username}:${option.password}@${option.host}:${option.port}/${option.database}`,
+      option
+    );
+    await sequelize.authenticate();
+
+    await sequelize.query(`CREATE SCHEMA ${testSchemaName};`);
+    const nodeConfig = new NodeConfig({
+      subquery: 'test',
+      dbSchema: testSchemaName,
+      proofOfIndex: true,
+      enableCache: false,
+      multiChain: true,
+    });
+    const project = {network: {chainId: '1'}, schema} as any;
+    const dbModel = new PlainStoreModelService(sequelize, nodeConfig);
+    storeService = new StoreService(sequelize, nodeConfig, dbModel, project);
+    await storeService.initCoreTables(testSchemaName);
+    await storeService.init(testSchemaName);
+    await storeService.modelProvider.metadata.set('startHeight', 1);
+    await storeService.modelProvider.metadata.set('lastProcessedHeight', 10000);
+    await storeService.modelProvider.metadata.set('lastProcessedBlockTimestamp', genBlockTimestamp(10000));
+
+    const chainId = '1';
+    const eventEmitter = new EventEmitter2();
+    multiChainRewindService = new MultiChainRewindService(
+      nodeConfig,
+      eventEmitter,
+      sequelize,
+      storeService,
+      mockBlockchainService as any
+    );
+
+    const reindex = jest.fn();
+
+    // Initialize the service
+    await multiChainRewindService.init(chainId, reindex);
+  });
+
+  afterEach(async () => {
+    await sequelize.query(`DROP SCHEMA ${testSchemaName} CASCADE;`);
+    multiChainRewindService.onApplicationShutdown();
+    await sequelize.close();
+  });
+
+  it('should handle rewind correctly', async () => {
+    // Act: Set global rewind lock to rewind to block 5
+    const rewindTimestamp = genBlockTimestamp(5);
+    await multiChainRewindService.setGlobalRewindLock(rewindTimestamp);
+
+    // Wait briefly for the PostgreSQL listener to process the notification
+    await delay(1);
+
+    // Assert: Check that the service is in Rewinding state and has the correct waitRewindHeader
+    expect(multiChainRewindService.status).toBe(RewindStatus.Rewinding);
```

Comment:
Reading the code, this doesn't need a delay, the status would be updated instantly.